tests/Microsoft.Bot.Builder.Tests/TurnContextTests.cs

```diff
@@ -35,6 +41,21 @@ public void Constructor()
             Assert.NotNull(c);
         }
 
+        [Fact]
+        public void TestTurnContextClone()
+        {
+            var c1 = new TurnContext(new SimpleAdapter(), new Activity() { Text = "one" });
+            c1.TurnState.Add("x", "test");
+            c1.OnSendActivities((context, activities, next) => next());
+            c1.OnDeleteActivity((context, activity, next) => next());
+            c1.OnUpdateActivity((context, activity, next) => next());
+            var c2 = new TurnContext(c1, new Activity() { Text = "two" });
+            Assert.Equal("one", c1.Activity.Text);
+            Assert.Equal("two", c2.Activity.Text);
+            Assert.Equal(c1.Adapter, c2.Adapter);
+            Assert.Equal(c1.TurnState, c2.TurnState);
```

Comment:
Can we assert what happens if `if (turnContext is TurnContext tc)` is true to make sure we cover that block too?